cli/src/test/java/org/jboss/gm/cli/DifferentJVMTest.java

```diff
@@ -49,38 +58,67 @@
     @Rule
     public TemporaryFolder tempDir = new TemporaryFolder();
 
-    private static final String jdkTar = "OpenJDK8U-jdk_x64_linux_hotspot_8u272b10.tar";
-    private static final String jdkDirectory = "jdk8u272-b10";
+    private static final Path GRADLE_JDK_HOME = Paths.get(System.getProperty("user.home"), ".gradle", "jdks");
 
-    private static final File gradleJDKHome = new File(
-            System.getProperty("user.home") + File.separator + ".gradle" + File.separator
-                    + "jdks");
-    private static final File jdk8 = new File(gradleJDKHome, jdkDirectory);
+    private static final String JDK8_BASEDIR = "jdk8u272-b10";
+
+    private static final Path JDK8_DIR = GRADLE_JDK_HOME.resolve(JDK8_BASEDIR);
+
+    private static final Path JDK8_BIN_DIR = JDK8_DIR.resolve("bin");
 
     /**
      * This method will, on Linux, download and cache if it doesn't exist a JDK8 installation from AdoptOpenJDK.
      *
-     * This location ($HOME/.gradle/jdks) was chosen to match https://docs.gradle.org/current/userguide/toolchains.html
+     * This location ($HOME/.gradle/jdks) was chosen to match https://docs.gradle.org/current/userguide/toolchains.html.
      * It utilises the same directory structure and location thereby matching potential prior downloads.
      */
     @BeforeClass
-    public static void setupJVM() {
-        if (SystemUtils.IS_OS_LINUX && !jdk8.exists()) {
-            //noinspection ResultOfMethodCallIgnored
-            gradleJDKHome.mkdirs();
-            File tarFile = new File(gradleJDKHome + File.separator + jdkTar);
-            File compressedFile = new File(tarFile + ".gz");
+    public static void setupJVM() throws IOException {
+        assumeTrue(SystemUtils.IS_OS_LINUX);
+
+        String filename = null;
+        UnArchiver ua = null;
+
+        if (SystemUtils.IS_OS_LINUX) {
+            filename = "OpenJDK8U-jdk_x64_linux_hotspot_8u272b10.tar.gz";
+            ua = new TarGZipUnArchiver();
+        } else if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Why are we testing for Windows if this is only run on Linux?

Reply:
I was actually trying to see if I could make it more generic, like if I had a `.zip` file and didn't have a `.tar.gz`. In the end, I found that I actually ran into a different issue.

1. The archive failed to extract, due to space or I don't know, but it creates the directory first, so the check for the directory passed, but there was no JDK in there. I slightly changed it to check for `bin/` instead, but that's not foolproof either.
2. I do not like that a test extracts a permanent JDK into my home directory, but this is a different issue that could perhaps be improved.
3. At least if we're extract to a known location, we could cache this in the CI so it doesn't need to download it each time.

Reply:
2/3) We should be caching in the CI ( I hope) ; as regards the local cache of a JDK it should be replicating the same as https://docs.gradle.org/current/userguide/toolchains.html

Reply:
I meant in the GitHub CI where we cache other build-related files between runs.

Reply:
+1 , I meant that and while I thought it was, perhaps it isn't from looking at it.

Reply:
I think the Gradle cache is something separate. I am not sure that it would cache the JDK itself.

For example, for Maven builds we might cache `.m2/repository` or anything else we download.

Reply:
But, Gradle cache may be a good idea nonetheless. I worry, though, as we seem to have "corrupted" artifacts in our caches that we had to manually remove.

Reply:
Let move the cache / windows discussion to a separate ticket (NCL-6679) to move this forward